### ISSUE TITLE
Fails to install on Safari v13.0

### ISSUE THREAD

**Issue by sans-github** [marioestrada/safari-omnikey#35]
Hello, 

I tired to install the plugin with Safari Version v(13.0 | 14608.1.49) on Mojave (v 10.14.6). It fails with following error message when I launch the extension file - 

Safari no longer supports this extension.
You can find new and updated extensions in the App Store or from developers’ websites.

Thanks, 
San

**Comment by sprdnja** [comment]
Same here. Any plans to support Safari 13?

**Comment by gkeramidas** [comment]
Apple removed support for what it calls "Legacy Extensions" in Safari 13 (anything using the .safariextz extension apparently). I don't know what is the recommended way of doing this in Safari 13, but it looks like it needs to be done through their App Store.

**Comment by gkeramidas** [comment]
Issue #34 seems to be the earlier version of this. I don't know if it's even possible to support Safari 13 anymore.

**Comment by poutena** [comment]
Just to add that I would also love to have a version of Omnikey compatible with Safari 13

**Comment by cylott** [comment]
@marioestrada Any plans to support the new app extension format? We love OmniKey!!

**Comment by Cartroo** [comment]
If you were willing and able to distribute it via the App Store, I'd be more than happy to pay $5-10 for this extension. I only upgraded yesterday and I'm already missing it!

(I realise it needs a number of people for this to be cost-effective, but perhaps it's worth investigating)

**Comment by cylott** [comment]
> Issue #34 seems to be the earlier version of this. I don't know if it's even possible to support Safari 13 anymore.

According to this comment, OmniKey is dead. I hope this isn't true.
https://github.com/marioestrada/safari-omnikey/issues/34#issuecomment-520680996

**Comment by RothOfKhan** [comment]
According to the developer of Safari Keyword Search, which is a very similar product "all new Safari extensions must use the new Safari App Extensions API. This new API does not include access to the address bar in the same way as the old API, which means that this extension cannot be ported"

I would assume this means that potential future support for Omnikey is dead.

**Comment by abey79** [comment]
If this is true, it is extremely sad. I've been using this type of shortcut for more than a decade on at least 3 different browsers.

Haste, on the App Store, addresses the same problem in a different way, which I obviously don't like (who likes change?). It's the only tool I found addressing this.

**Comment by RothOfKhan** [comment]
The app Alfred also addresses the problem in a different manner and is the closest thing I've found to being able to mimic the behavior of Omnikey. I do prefer it to Haste's attempt, as it seems more intuitive, but it isn't perfect. Alfred also comes with a lot of other features that I've used in the past, so I'm trying to decide if I want to just learn the new workflow or if I want to change browsers across all my devices. 

Another option is using duckduckgo's !bangs feature, but it is less customizable.

**Comment by abey79** [comment]
@RothOfKhan Thanks for the pointer. I installed Alfred and it is just great. It should not take too long to adjust from `cmd-L` to `cmd-space` (which is what I set it up with) before typing my custom searches. I was heavily using Spotlight as app launcher and file system navigation, so the transition is quite natural.

This means I can free myself from current and future shenanigans from Apple with Safari plugin API, which is nice.

**Comment by supertopher** [comment]
Yes i'm also annoyed at apple.  If anyone gets excited [here is the apple guide for conversion to the app store](https://developer.apple.com/documentation/safariservices/safari_app_extensions/converting_a_legacy_safari_extension_to_a_safari_app_extension)

**Comment by duckquack** [comment]
> Yes i'm also annoyed at apple. If anyone gets excited [here is the apple guide for conversion to the app store](https://developer.apple.com/documentation/safariservices/safari_app_extensions/converting_a_legacy_safari_extension_to_a_safari_app_extension)
Right, but the issue seems to be there is no way to access the address bar.

**Comment by duckquack** [comment]
> Yes i'm also annoyed at apple. If anyone gets excited [here is the apple guide for conversion to the app store](https://developer.apple.com/documentation/safariservices/safari_app_extensions/converting_a_legacy_safari_extension_to_a_safari_app_extension)

We have yet to hear from @marioestrada but it seems like the issue is the new app extension model won't support address bar filtering.

I'm using Haste and it is okay, but a much less elegant solution. I will miss OmniKey. I would have happily paid $10 for it.

**Comment by abey79** [comment]
I have now completely switched to Alfred as per @RothOfKhan suggestion, and I'm happy about it. I find this solution superior to Haste because it piggy-backs on an existing habit (be it Spotlight, LaunchBar, or any other launcher you may be using) instead of introducing yet another entry point as Haste does. Of  course, YMMV.

**Comment by duckquack** [comment]
> I have now completely switched to Alfred as per @RothOfKhan suggestion, and I'm happy about it. I find this solution superior to Haste because it piggy-backs on an existing habit (be it Spotlight, LaunchBar, or any other launcher you may be using) instead of introducing yet another entry point as Haste does. Of course, YMMV.

I'm not looking for a bloated swiss army knife type app, just a tool to expand URL shortcuts. This is what was great about OmniKey, it was simple and elegant. Thanks for nothing, Apple.

**Comment by Spirarel** [comment]
[This looks promising](https://macscripter.net/viewtopic.php?id=47107)

**Comment by trocotronic** [comment]
> [This looks promising](https://macscripter.net/viewtopic.php?id=47107)

But how it is used?

**Comment by Spirarel** [comment]
Type in your search just like normal in the search bar - > execute the
Applescript

You'll want to make a keybinding for the later.

**Comment by gitcabezon** [comment]
Here's an option for Keyboard Maestro (KM) users --

Copy/paste this into Script Editor:

`

var myQueries = { a:"http://www.amazon.com/s/?link_code=wsw&_encoding=UTF-8&search-alias=aps&field-keywords={search}&Submit.x=0&Submit.y=0&Submit=Go", y:"http://www.youtube.com/results?search_query={search}", e:"http://www.ebay.com/sch/i.html?_trksid=p5197.m570.l1313&_nkw={search}&_sacat=0", w:"http://en.wikipedia.org/w/index.php?title=Special:Search&search={search}", g:"https://www.google.com/search?&q={search}", gm:"https://maps.google.com/maps?hl=en&authuser=0&q={search}&ie=UTF-8", imdb:"http://www.imdb.com/find?q={search}&s=all"}

var app = Application("System Events")
app.includeStandardAdditions = true
 
var dialogText = "Enter omni-key-style shortcut:"
var response = app.displayDialog(dialogText, {
    defaultAnswer: "",
    withIcon: "note",
    buttons: ["Cancel", "Continue"],
    defaultButton: "Continue"
})

var myQueryList = response["textReturned"].split(" ")
var myQueryType = myQueryList[0]
myQueryList.shift()
var myQuery2 = myQueryList.join("%20")

var myStr2 = myQueries[myQueryType].replace("{search}", myQuery2)

app.openLocation(myStr2)
`

... then set the language as JavaScript and save as a .scpt file into ~/Library/Scripts/Applications/Safari.  Then, create a KM macro that only works within Safari that (a) types the keystroke "cmd-l" (ell), and (b) runs that script.  If you just want to be in the location bar hit escape after hitting cmd-l.

Wish OmniKey still worked, but this works half-decently.

**Comment by cylott** [comment]
> Here's an option for Keyboard Maestro (KM) users --
> 
> Copy/paste this into Script Editor:
> 
> `
> 
> var myQueries = { a:"[http://www.amazon.com/s/?link_code=wsw&_encoding=UTF-8&search-alias=aps&field-keywords={search}&Submit.x=0&Submit.y=0&Submit=Go](http://www.amazon.com/s/?link_code=wsw&_encoding=UTF-8&search-alias=aps&field-keywords=%7Bsearch%7D&Submit.x=0&Submit.y=0&Submit=Go)", y:"[http://www.youtube.com/results?search_query={search}](http://www.youtube.com/results?search_query=%7Bsearch%7D)", e:"[http://www.ebay.com/sch/i.html?_trksid=p5197.m570.l1313&_nkw={search}&_sacat=0](http://www.ebay.com/sch/i.html?_trksid=p5197.m570.l1313&_nkw=%7Bsearch%7D&_sacat=0)", w:"[http://en.wikipedia.org/w/index.php?title=Special:Search&search={search}](http://en.wikipedia.org/w/index.php?title=Special:Search&search=%7Bsearch%7D)", g:"[https://www.google.com/search?&q={search}](https://www.google.com/search?&q=%7Bsearch%7D)", gm:"[https://maps.google.com/maps?hl=en&authuser=0&q={search}&ie=UTF-8](https://maps.google.com/maps?hl=en&authuser=0&q=%7Bsearch%7D&ie=UTF-8)", imdb:"[http://www.imdb.com/find?q={search}&s=all"}](http://www.imdb.com/find?q=%7Bsearch%7D&s=all%22%7D)
> 
> var app = Application("System Events")
> app.includeStandardAdditions = true
> 
> var dialogText = "Enter omni-key-style shortcut:"
> var response = app.displayDialog(dialogText, {
> defaultAnswer: "",
> withIcon: "note",
> buttons: ["Cancel", "Continue"],
> defaultButton: "Continue"
> })
> 
> var myQueryList = response["textReturned"].split(" ")
> var myQueryType = myQueryList[0]
> myQueryList.shift()
> var myQuery2 = myQueryList.join("%20")
> 
> var myStr2 = myQueries[myQueryType].replace("{search}", myQuery2)
> 
> app.openLocation(myStr2)
> `
> 
> ... then set the language as JavaScript and save as a .scpt file into ~/Library/Scripts/Applications/Safari. Then, create a KM macro that only works within Safari that (a) types the keystroke "cmd-l" (ell), and (b) runs that script. If you just want to be in the location bar hit escape after hitting cmd-l.
> 
> Wish OmniKey still worked, but this works half-decently.

You could also set this up in Tampermonkey.. oh wait, Apple killed that too.

**Comment by DavidMunch** [comment]
For what it is worth, I also switched to Alfred. Very happy with it, and I think it will integrate quite nicely in my workflow, and might even expand it since it is so capable.

**Comment by Jimmbo** [comment]
Hey guys, 

I have a tip for improving workflow with Alfred. 

First, it should be stressed that Alfred works EXACTLY the same way as OmniKey...except that instead of typing into Safari's URL field, you type into Alfred's launch field. Plus Alfred offers a few advantages. You can add site logos, test your shortcuts, and encode spaces in different ways.

Ok, here's the tip: Alfred defaults to being launched via option/space. But you can change this.  And if you, like me and abey79 (per above), are accustomed to launching OmniKey by going to Safari and typing Command/L to select the URL field, here's a hot tip: just set Alfred to launch via Command/L. Congratulations: you have now IMPROVED on OmniKey. You don't even need to be in Safari to launch a keyword search. Even if you're in, say, TextEdit, type Command/L and enter your site abbreviation and search terms and Safari will open with results (note: apps currently using Command/L will be steamrolled by Alfred, so you may want to reassign new shortcuts to those functions).

Only issue: if you're in Safari, you need to accept that Command/L will pop up Alfred. But you can train Alfred to do whatever URL stuff you wanted Safari to do.

One note. Abey79, above,  mentions using Spotlight for app launching. I use it for seldom-run apps (though you can just as easily use Alfred the exact same way). But for very frequently used apps, you can't beat launching via keyboard shortcuts, and the paid Alfred powerpack lets you assign keyboard shortcuts to launch apps, folders, or scripts, and functions like sleep or restart. I highly recommend this. I can zoom from Safari to Stickies to TextEdit to Mail and back in 3 seconds flat.

**Comment by Jimmbo** [comment]
Does anyone know where OmniKey prefs live? I'd like to get at my previous table of shortcuts/sites, just so I can config Alfred efficiently.

**Comment by Jimmbo** [comment]
Me again. 

To address the issue that configuring command-L to launch Alfred means you'll launch Alfred even while in Safari (i.e. losing the ability to conveniently go to URL bar):

Just train Alfred to make Safari do whatever you'd normally want done in the URL bar. 

For example, I'm a big fan of the Twitter feed of Rick Wilson, so I often type Command-L, then type Ri, and Safari knows what I want, expands the URL, and goes to the right page. 

So....I bookmarked his Twitter page, went to Alfred Prefs / Web Bookmarks and checked "Safari Bookmarks", so Alfred will look in Safari's bookmarks for matches to whatever you type.  Next step: I called up Alfred (Command-L), typed "Ri", and it presented many options, one of which was the correct one. I chose the correct one, and Alfred immediately learned my pref (it's now the default action). 

Repeat for all similar favorite sites, and Alfred will behave transparently as Safari's URL bar always behaved...whether doing site searches (i.e. via OmniKey) or launching favorite web pages via text strings. ZERO functionality lost, and a better experience, generally (Alfred's way smarter than Safari ever was).

**Comment by DavidMunch** [comment]
For what it is worth, it took me under a week to settle in comfortable with Alternate-Space for starting Alfred, despite having used Command-L for so many years. Being able to trigger a search anywhere in the OS and not just in Safari is just icing on the cake. And the expanded information I get from Alfred makes it even batter.

I'm a happy switcher, and even if Omnikey comes back, I likely won't be using it. :) So thank you to @RothOfKhan for the recommendation!

**Comment by RothOfKhan** [comment]
I've also - after suggesting it a few weeks ago - settled into using Alfred. 

Obviously, the cost can be prohibitive for some users, but I'm also really liking the Power Pack as an addition as well. [Searchio](https://github.com/deanishe/alfred-searchio/releases) allows me to get search suggestions for almost all of the major search sites I use, while the [Dark Sky](https://github.com/kejadlen/dark-sky.alfredworkflow/releases) workflow allows weather information to populate directly in the Alfred search bar, and the [SpeedTest](https://github.com/chusiang/alfred-workflows/tree/master/speedtest) workflow queries up a speedtest.net test without having to visit the site. There's a lot of these available and I'm currently using 20 of them right now to do various tasks. 

Without the power pack, I still feel that Alfred would be a full replacement for Omnikey, even if it does change the workflow a bit - but the Power Pack is taking it to a level that Omnikey was never able to reach. You can search on alfredforum.com and packal.org if you're interested in the workflows to see if there are any that could help you.

**Comment by Spirarel** [comment]
I was completely avoiding Alfred because it's not the fix I wanted, but after binding it to `CMD, L, L` in Safari, it's not the worst substitute.

It's definitely not the same, you can't open a tab in the background with this method and every search spawns a new tab, but it's something.

Best of luck coping, fellow power-users.

**Comment by Cartroo** [comment]
As another data point, I was avoiding Alfred at first, partly as I thought a separate app wouldn't be convenient enough and partly because getting approval to run new apps at work is a chore. I noticed someone else had already had it approved, though, so I gave it a shot and I'm finding it's a very competent Ominkey substitute even without the paid upgrade (I haven't yet decided if I'm going to try that).

I agree that it's behaviour within the browser is less flexible, and it does have a funny habit of popping up new windows instead of tabs occasionally (I think it's when I closed a Safari window and hadn't focused another one). I find these minor niggles as I'm used to juggling tabs between multiple browser windows, as a lot of apps open them in the "wrong" one for my liking, but I can see how it would be a major annoyance for some people.

Also, I already use Things 3 on my Mac, however, which binds CTRL-SPACE, so it may be that my muscle-memory is already half-trained for additional *-SPACE shortcuts.

**Comment by marioestrada** [comment]
Hey all currently I haven't found time to look into updating the Extension for Catalina and latest Safari. I hope to find some time to look into alternatives in the next month or so.

**Comment by cylott** [comment]
> Hey all currently I haven't found time to look into updating the Extension for Catalina and latest Safari. I hope to find some time to look into alternatives in the next month or so.

Thanks for the update. If you do find a way to accomplish address bar filtering with the new API's, please charge money for it. The alternatives suck.

**Comment by gkeramidas** [comment]
If this works with address bar filtering in the future, assuming such a mechanism exists, I am also happy to pay for this extension. Thanks for your work making this. It’s absolutely great!

**Comment by KontiKonsumi** [comment]
Would also be willing for paying for it if its inside Safari

**Comment by cylott** [comment]
Ok, well it is official. CustomSearch now provides all the functionality that OmniKey once did. With the latest update, it is no longer necessary to prepend a search with a "!"

Switch to CustomSearch. It has an ugly icon, but it works perfectly.
https://sidetree.com/customsearch.html

Thanks Apple for creating this clumsy app extension model, so that we can all pay devs a bounty to recreate what already worked fine. I guess that's how "value" is created.

**Comment by gitcabezon** [comment]
CustomSearch looks good, but why is it $10?  Seems like a lot for such a narrow feature set when something like BBEdit is $39.

**Comment by DavidMunch** [comment]
Because it is the only app on the market providing the function, I would wager.

**Comment by cylott** [comment]
> CustomSearch looks good, but why is it $10? Seems like a lot for such a narrow feature set when something like BBEdit is $39.

This is my point exactly. You are paying a dev a "bounty" to go to the trouble of jumping through Apple's 1000 hoops.

**Comment by gitcabezon** [comment]
Something very close can be done for free with AppleScript and a keyboard macro program like Keyboard Maestro, so it's not saving people that much.
